internal/kubernetes/operator/install_resources.go

```diff
@@ -199,17 +199,19 @@ func (ir *InstallResources) addServiceAccount(ctx context.Context, config map[st
 }
 
 func (ir *InstallResources) addRoles(ctx context.Context, config map[string]interface{}, namespace string, watch []string) error {
-	namespaces := []string{namespace}
+	namespaces := map[string]struct{}{namespace: {}}
 
 	if value, ok := config["metadata"]; ok {
 		if metadata, ok := value.(map[string]interface{}); ok {
 			if name, ok := metadata["name"]; ok && name != leaderElectionRoleName {
-				namespaces = append(namespaces, watch...)
+				for _, watchedNamespace := range watch {
+					namespaces[watchedNamespace] = struct{}{}
+				}
```

Comment:
[nit] this is level of nesting is starting to smell 👃, consider ways to simplify the logic, in particular consider nested `if`scould be simplified as a `&&` or you could/should introduce functions

Reply:
Fair point. I also didn't like it much. I have refactored the two places